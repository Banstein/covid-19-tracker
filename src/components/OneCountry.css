.card {
  background: var(--secondary);
  padding: 5px;
  position: relative;
  text-align: center;
  height: 100%;
}

.card:hover {
  border: 1px solid var(--terciary);
}

.fa-circle-right {
  font-weight: 100;
  position: absolute;
  right: 10px;
  top: 10px;
}

.map {
  opacity: 0.4;
}
