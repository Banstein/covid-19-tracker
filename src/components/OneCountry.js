/* eslint-disable camelcase */
import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import './OneCountry.css';

function OneCountry({ name, today_confirmed }) {
  return (
    <NavLink to={`/${name}`}>
      <li className="card">
        <div>
          <img className="map" src="https://raw.githubusercontent.com/djaiss/mapsicon/33ba28808f8d32b5bae0ffada9cadd07073852e1/all/ad/vector.svg" alt="map" />
          <i className="fa-regular fa-circle-right" />
        </div>
        <h5>{name}</h5>
        <h6>{today_confirmed}</h6>
      </li>
    </NavLink>
  );
}

OneCountry.propTypes = {
  name: PropTypes.string.isRequired,
  today_confirmed: PropTypes.number.isRequired,
};

export default OneCountry;
