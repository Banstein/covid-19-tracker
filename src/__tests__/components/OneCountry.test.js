/* eslint-disable camelcase */
import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import store from '../../__mocks__/configureStore';
import OneCountry from '../../components/OneCountry';

describe('One Country snapshot', () => {
  const id = 'germany';
  const name = 'Germany';
  const today_confirmed = 47;
  test('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <OneCountry id={id} name={name} today_confirmed={today_confirmed} />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
