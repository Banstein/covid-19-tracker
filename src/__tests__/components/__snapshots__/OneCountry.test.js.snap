// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`One Country snapshot renders correctly 1`] = `
<a
  className=""
  href="/Germany"
  onClick={[Function]}
>
  <li
    className="card"
  >
    <div>
      <img
        alt="map"
        className="map"
        src="https://raw.githubusercontent.com/djaiss/mapsicon/33ba28808f8d32b5bae0ffada9cadd07073852e1/all/ad/vector.svg"
      />
      <i
        className="fa-regular fa-circle-right"
      />
    </div>
    <h5>
      Germany
    </h5>
    <h6 />
  </li>
</a>
`;
